en/advanced-topics/jsb-auto-binding.md

```diff
@@ -0,0 +1,375 @@
+
+Fixing this problem requires a JSB modification for log calls, along with a caching mechanism related to jni to optimize performance. jSB binding is simply the process of converting objects between the C++ and script layers, and forwarding script layer function calls to the C++ layer.
+
+JSB binding is usually done in two ways: **manual binding** and **automatic binding**. The manual binding method can be found in the [Using JSB Manual Binding](jsb-manual-binding.md) documentation.
+- The advantage of manual binding is that it is flexible and customizable; the disadvantage is the necessity to write all the code yourself, especially the conversion between C++ type and TypeScript type, which can easily lead to memory leaks and some pointers or objects not being freed.
+- Auto-binding will save a lot of trouble, which directly generates the relevant code through a script in one click, and subsequently, if there are new codes or changes, simply re-execute the script once. Auto-binding is perfect for cases where it is not necessary to do strong customization and then need to finish JSB quickly. The following is a step-by-step explanation of how to implement automatic JSB binding.
+
+## Environment Configuration and Auto-Binding Demonstration
+
+### Environment Configuration
+
+Auto-binding, to put it simply, is a matter of executing a python script to automatically generate the corresponding `.cpp` and `.h` files. First,make sure that the computer has a python runtime environment. Example of how to install it on a Mac:
+
+1. To install python 3.0, download the installation package from the python website:
+
+    <https://www.python.org/downloads/release/python-398/>
+
+2. Install some of the python dependencies via pip3:
+
+    ```shell
+    sudo pip3 install pyyaml==5.4.1
+    sudo pip3 install Cheetah3
+    ```
+
+3. Install the NDK, which is definitely essential when it comes to C++. It is recommended to install the [Android NDK r16b](https://developer.android.com/ndk/downloads/older_releases?hl=zh-cn) version, set the `PYTH_profile` in `~/.bash_ profile`, then set `PYTHON_ROOT` and `NDK_ROOT` in `~/.bash_profile`, because these two environment variables will be used directly in the python file that will be executed later.
```

Comment:
```suggestion
3. Install the NDK, which is definitely essential when it comes to C++. It is recommended to install the [Android NDK r16b](https://github.com/android/ndk/wiki/Unsupported-Downloads#r16b) version, set the `PYTH_profile` in `~/.bash_ profile`, then set `PYTHON_ROOT` and `NDK_ROOT` in `~/.bash_profile`, because these two environment variables will be used directly in the python file that will be executed later.
```